### ISSUE TITLE
Policy statements operator returns empty arrays (stub implementation)

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#104]
## Problem
The \`opPolicyStatements\` operator is a stub that returns empty arrays, marked with TODO comments.

## Location
\`go/pkg/dtrules/operators/stack.go:85-99\`

## Current Code
```go
// TODO: Implement proper policy statement handling
func opPolicyStatements(state dtrules.State) error {
    arr, err := dtrules.NewArray(state.GetSession(), true, false)
    // ...returns empty array...
}
```

## Impact
Policy statement tracking not available. Decision tables cannot report which policy statements were triggered.

## Suggested Fix
Implement proper policy statement collection during table execution:
1. Track policy statements in State when columns are executed
2. Return collected statements when operator is called

## Priority
HIGH - Affects audit trail and policy reporting.

**Comment by PaulSnow** [comment]
Fixed: Implemented proper policy statement collection in opPolicyStatements operator.
